### PR TITLE
Add tag for DG formulation

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

Add an option tag and tag so we can store the DG formulation in the DataBox.

Note that I add a `dg::Tags` namespace so I have to add `::` in a bunch of places that use `::Tags` but are inside the `dg` namespace.

Depends on:
- [x] #2548 

Only the following commits are new:
- Add tags for dg::Formulation (with 167 additions and 22 deletions.)

### Types of changes:

- [ ] Bugfix
- [ ] New feature
- [ ] Refactor

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Comment by nilsdeppe** [comment]
Did some silly CMake stuff that I need to clean up before this should be looked at...

**Comment by nilsdeppe** [comment]
I've sorted out my CMake goofs...

File: tests/Unit/NumericalAlgorithms/DiscontinuousGalerkin/Tags/Test_Formulation.cpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+#include "Helpers/DataStructures/DataBox/TestHelpers.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/Formulation.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/Tags/Formulation.hpp"
+
+SPECTRE_TEST_CASE("Unit.DiscontinuousGalerkin.Tags.Formulation",
+                  "[Unit][Evolution]") {
+  TestHelpers::db::test_simple_tag<dg::Tags::Formulation>("Formulation");
+  CHECK(
+      TestHelpers::test_creation<dg::Formulation, dg::OptionTags::Formulation>(
+          "Strong") == dg::Formulation::StrongInertial);
```
Comment (fmahebert) [inline tests/Unit/NumericalAlgorithms/DiscontinuousGalerkin/Tags/Test_Formulation.cpp]: I would strongly prefer for the creation string to match the actual name of the enum -- having these differ is just inviting confusion down the line. I don't care if you rename the string or the enum :)

File: src/NumericalAlgorithms/DiscontinuousGalerkin/Tags/Formulation.hpp [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+/*!
+ * \brief The DG formulation to use.
+ */
+struct Formulation : db::SimpleTag {
+  using type = dg::Formulation;
+
+  using option_tags = tmpl::list<OptionTags::Formulation>;
+  static constexpr bool pass_metavariables = false;
+  static dg::Formulation create_from_options(
+      const dg::Formulation& quadrature) noexcept;
```
Comment (fmahebert) [inline src/NumericalAlgorithms/DiscontinuousGalerkin/Tags/Formulation.hpp]: `quadrature` -> `formulation` ?

File: src/NumericalAlgorithms/DiscontinuousGalerkin/Tags/Formulation.cpp [hunk 3]
```diff
@@ -0,0 +1,13 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "NumericalAlgorithms/DiscontinuousGalerkin/Tags/Formulation.hpp"
+
+#include "NumericalAlgorithms/DiscontinuousGalerkin/Formulation.hpp"
+
+namespace dg::Tags {
+dg::Formulation Formulation::create_from_options(
+    const dg::Formulation& quadrature) noexcept {
```
Comment (fmahebert) [inline src/NumericalAlgorithms/DiscontinuousGalerkin/Tags/Formulation.cpp]: `quadrature` -> `formulation` ?

**Review by fmahebert — CHANGES_REQUESTED** [review]

**Comment by nilsdeppe** [comment]
@fmahebert I forgot to push the fixup... I rebased too. Thanks for the review!

**Review by fmahebert — COMMENTED** [review]
I am happy with fixup, please squash

**Review by kidder — COMMENTED** [review]
I have no change requests

**Comment by nilsdeppe** [comment]
Squashed and rebased on develop. Thanks for the reviews!

**Review by kidder — DISMISSED** [review]

**Review by fmahebert — DISMISSED** [review]

**Comment by nilsdeppe** [comment]
Rebased on develop to resolve conflicts., no longer dependent

**Review by kidder — APPROVED** [review]

**Review by fmahebert — APPROVED** [review]

**Pull request merged**